### PR TITLE
TES-8356 support AWS SDK v3

### PR DESCRIPTION
**Pull request by liorama**
Use the new `@aws-sdk/lib-storage` instead of the old S3 client's `upload` method.

\* This PR is branched from RonK's commit `bac61942629d15c8bae86eae1cc550cf39ddf7e8` (which builds off of two other branches) that is currently used in services but compared with master (hence the two extra unrelated commits).

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -3,8 +3,9 @@ var stream = require('stream')
 var fileType = require('file-type')
 var isSvg = require('is-svg')
 var parallel = require('run-parallel')
+var Upload = require('@aws-sdk/lib-storage').Upload;
```
Comment (Linkgoron) [inline index.js:6]:
```suggestion
const { Upload } = require('@aws-sdk/lib-storage');
```

File: index.js [hunk 2]
```diff
@@ -179,31 +200,49 @@ S3Storage.prototype._handleFile = function (req, file, cb) {
...
         storageClass: opts.storageClass,
         serverSideEncryption: opts.serverSideEncryption,
         metadata: opts.metadata,
         location: result.Location,
         etag: result.ETag,
         versionId: result.VersionId
       })
-    })
-  })
+    }).catch(cb);
```
Comment (Linkgoron) [inline index.js:243]: I assume that you don't want the cb to get called twice if it throws synchronously. Either move the `catch` before the `then`, or use `.then(xxx,yyy)` syntax.

File: index.js [hunk 3]
```diff
@@ -179,31 +200,49 @@ S3Storage.prototype._handleFile = function (req, file, cb) {
...
+        params: params,
+      })
 
+      upload.on('httpUploadProgress', function (progress) {
+        if (progress.total) {
+          currentSize = progress.total;
+        }
+      });
+    } catch (e) {
+      cb(e);
```
Comment (Linkgoron) [inline index.js:223]: should probably return here, and not call done.

File: index.js (hunk 1 above)
Comment (liorama) [reply index.js:6]: I know our runtime supports this but the code currently and all past changes we've made are ES5 compatible, I didn't want to break that.

File: index.js [hunk 4]
```diff
@@ -179,31 +200,50 @@ S3Storage.prototype._handleFile = function (req, file, cb) {
...
+      return;
+    }
+    
+    upload.done().catch(cb).then(function (result) {
```
Comment (Linkgoron) [inline index.js:227]: this is actually a bad solution, and was a bad suggestion by me. Now it'll always call cb twice when an error happens. I guess that the best solution would be to use then(successPath,failurePath)

**Comment by liorama** [comment]
@Linkgoron Are you ok with merging this?

**Comment by Linkgoron** [comment]
yes

**Review by Linkgoron — APPROVED** [review]

**Pull request merged**